extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/DevServicesElasticsearchProcessor.java

```diff
@@ -0,0 +1,180 @@
+import io.quarkus.devservices.common.ConfigureUtil;
+import io.quarkus.devservices.common.ContainerAddress;
+import io.quarkus.devservices.common.ContainerLocator;
+import io.quarkus.runtime.configuration.ConfigUtils;
+
+/**
+ * Starts an Elasticsearch server as dev service if needed.
+ */
+public class DevServicesElasticsearchProcessor {
+    private static final Logger log = Logger.getLogger(DevServicesElasticsearchProcessor.class);
+    private static final String ELASTICSEARCH_HOSTS = "quarkus.elasticsearch.hosts";
+
+    /**
+     * Label to add to shared Dev Service for Elasticsearch running in containers.
+     * This allows other applications to discover the running service and use it instead of starting a new instance.
+     */
+    static final String DEV_SERVICE_LABEL = "quarkus-dev-service-elasticsearch";
+    static final int ELASTICSEARCH_PORT = 9200;
+
+    private static final ContainerLocator elasticsearchContainerLocator = new ContainerLocator(DEV_SERVICE_LABEL,
+            ELASTICSEARCH_PORT);
+
+    static volatile DevServicesResultBuildItem.RunningDevService devService;
+    static volatile ElasticsearchDevServicesBuildTimeConfig cfg;
+    static volatile boolean first = true;
+
+    private final IsDockerWorking isDockerWorking = new IsDockerWorking(true);
+
+    @BuildStep(onlyIfNot = IsNormal.class, onlyIf = GlobalDevServicesConfig.Enabled.class)
+    public DevServicesResultBuildItem startElasticsearchDevService(
+            LaunchModeBuildItem launchMode,
+            ElasticsearchDevServicesBuildTimeConfig configuration,
+            List<DevServicesSharedNetworkBuildItem> devServicesSharedNetworkBuildItem,
+            Optional<ConsoleInstalledBuildItem> consoleInstalledBuildItem,
+            CuratedApplicationShutdownBuildItem closeBuildItem,
+            LoggingSetupBuildItem loggingSetupBuildItem, GlobalDevServicesConfig devServicesConfig) {
+
+        if (devService != null) {
+            boolean shouldShutdownTheBroker = !configuration.equals(cfg);
+            if (!shouldShutdownTheBroker) {
+                return devService.toBuildItem();
+            }
+            shutdownElasticsearch();
+            cfg = null;
+        }
+
+        StartupLogCompressor compressor = new StartupLogCompressor(
+                (launchMode.isTest() ? "(test) " : "") + "Elasticsearch Dev Services Starting:",
+                consoleInstalledBuildItem, loggingSetupBuildItem);
+        try {
+            devService = startElasticsearch(configuration, launchMode,
+                    !devServicesSharedNetworkBuildItem.isEmpty(),
+                    devServicesConfig.timeout);
+            compressor.close();
+        } catch (Throwable t) {
+            compressor.closeAndDumpCaptured();
+            throw new RuntimeException(t);
+        }
+
+        if (devService == null) {
+            return null;
+        }
+
+        // Configure the watch dog
+        if (first) {
+            first = false;
```

Comment:
Strictly speaking, this is not atomic and could result in the watchdog being registered multiple times.

But I see we use this (unsafe) code in other devservices, so I suppose it's no big deal to do this once more.

Reply:
Yeah, it's copy/paste code.

I can have a look and ask Stuart but build steps are run in a specific mode so it may be safe.